tests/test_resource.py

```diff
@@ -484,8 +484,8 @@ def test_method(self) -> Response:
 
         XResource(event)()
 
+        mocked_event_api_clear.assert_called_once()
```

Comment:
This is not an ideal test as the order of running these methods matters 😐 

I think we need to resign from this way of mocking and test it more directly by checking the list of events.